QFT/sample_solutions/evaluation.py

```diff
@@ -0,0 +1,64 @@
+def evaluate_patient(patient_data):
+    """Decide on a patient's test result.
+
+    Takes as its argument a sequence of four elements interpreted as
+    the measured IFN-gamma levels from a single patient in the following
+    order:
+    test sample 1, test sample 2, negative control, positive control
+
+    Returns:
+      1  if the data suggests a POSITIVE test result,
+      0  if the data suggests a NEGATIVE test result,
+     -1  if the data is inconclusive
+    """
+    
+    # lesbare Benennung aller im weiteren benötigten Werte
+    neg_control = patient_data[2]
+    delta1 = patient_data[0] - neg_control
+    delta2 = patient_data[1] - neg_control
+    delta_pos = patient_data[3] - neg_control
+
+    # Entscheidungsbaum
+    if neg_control > 8:
+        return -1
+    if delta1 >= 0.35 and 100*delta1/neg_control >= 25:
+        return 1
+    if delta2 >= 0.35 and 100*delta2/neg_control >= 25:
+        return 1
+    if delta_pos < 0.5:
+        return -1
+    return 0
+
+
+def test_it(patient_data, expect):
+    print('Testing with', patient_data, '...', end=' ')
+    result = evaluate_patient(patient_data)
+    if  result == expect:
+        print('passed.')
+    else:
+        print('FAILED!', 'Expected', expect, ', got', result, '.')
+
+
+if __name__ == '__main__':
+    test_data_sets = [
```

Comment:
This prepares a list of test datasets and expected results to be passed to the `test_it()` function.
Note that this is a ``list`` of ``tuples``, where each ``tuple`` represents a *test case* and consists of a patient data tuple and the corresponding return value expected from `evaluate_patient()`.

Indentation and line breaks are irrelevant inside the square brackets of the outer list and we can use this to our advantage to make the code more readable.